### DISCUSSION TITLE
Cluster with uWebSocket

### DISCUSSION THREAD

**Discussion by yovanoc** [uNetworking/uWebSockets.js#220] [Q&A] [answered]
Hello, follow up of my message on #214 I just want to know how to make it work because when I done that:

```ts
import cluster from "cluster";
import { cpus } from "os";

const numCPUs = cpus().length;

function masterProcess() {
  console.log(`Master ${process.pid} is running`);

  for (let i = 0; i < numCPUs; i++) {
    console.log(`Forking process number ${i}...`);
    cluster.fork();
  }
}

function childProcess() {
  console.log(`Worker ${process.pid} started...`);

  app
    .any("/*", res => {
      /* Let's deny all Http */
      res.end("Nothing to see here! (" + process.pid + ")");
    })
    .listen(serverPort, listenSocket => {
      if (listenSocket) {
        log(`Listening to port ${serverPort}`);
      }
    });
}

if (cluster.isMaster) {
  masterProcess();
} else {
  childProcess();
}

```

I always get the same PID, I don't know why, I just tested with a simple httpServer include with node and it works, so can it really works like that?

![image](https://user-images.githubusercontent.com/7302317/67848304-aa78df80-fb04-11e9-9891-777de5032f88.png)

Sorry for my noobiness :x

**Comment by ghost** [comment]
I get different pid. You probably get the same because your browser keeps the same connection. Try with curl and you'll see different pids.

**Comment by yovanoc** [comment]
![68498326-de67a980-0256-11ea-8d09-ac20dc93a40d](https://user-images.githubusercontent.com/7302317/68500920-24c00700-025d-11ea-884f-5c7945b0e031.png)

I've already test it.... so i'ts my computer who give me always the same because is not "stressed" enough to load balance to another?

**Comment by yovanoc** [comment]
Sorry to comment again but is the `app.listen` function who have to be in the `childProcess ` or all app.get, app.ws, etc too? thanks

**Comment by yovanoc** [comment]
Sorry @alexhultman , but can I have an answer on my last question?

**Comment by ghost** [comment]
Everything has to be in child, the entire app should be isolated per process or per thread

**Comment by yovanoc** [comment]
Yes I see now why, great thanks, but I really don't know how to do this with a clean architecture since I have a lot of routes http and ws

**Comment by taehyeong224** [comment]
I got the same PID with a similar code in macOS Catalina. but I got the different PID in ubuntu 16.04.

**Comment by ghost** [comment, marked as answer]
Clustering is a Linux feature, it does not work on Windows or macOS. You'll get different PIDs when you stress the server on Linux.

**Comment by animaonline** [comment]
@alexhultman forking the process several times in order to "create a cluster" is pretty common in nodejs, it works on all platforms, on Linux and macOS TCP sockets are shared across forked instances, it works with standard tcp sockets / express and other frameworks, is there a reason why it doesn't work with uWebSockets.js? Does it need to be implemented in the native module?

**Reply by e3dio** [reply]
Open issue here https://github.com/uNetworking/uWebSockets/issues/1194 I would use the windows cluster if avail, maybe just needs some effort

**Reply by animaonline** [reply]
@e3dio thanks man
